### PR TITLE
Issue #993 fix: shareViaMail methods now shows only mail apps

### PR DESCRIPTION
**Pull request by ShuvranilSaha**
shareViaMail methods now shows only mail apps or list of mail apps in which user wants to choose and share

### CODE REVIEW

**Comment by EddyVerbruggen** [comment]
I can impossibly merge this.. can you please check https://github.com/EddyVerbruggen/SocialSharing-PhoneGap-Plugin/pull/994/files (~730 lines changed!) and only send a PR for the actual lines you had to change to make it work?

**Comment by subranil** [comment]
Sure, doing in right now and updating it

**Comment by subranil** [comment]
@EddyVerbruggen if you can check , clarify and merge.. it would be great. because we are using that plugin on our project.

**Comment by EddyVerbruggen** [comment]
Thank you, this seems to work really well! Released 5.4.2 just now.

File: src/android/nl/xservices/plugins/SocialSharing.java [hunk 1]
```diff
@@ -183,12 +183,26 @@ public void run() {
         // this was added to start the intent in a new window as suggested in #300 to prevent crashes upon return
         draft.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
 
-        draft.setType("application/octet-stream");
+        draft.setData(Uri.parse("mailto:"));
+
+                List<ResolveInfo> emailAppList = cordova.getActivity().getPackageManager().queryIntentActivities(draft, 0);
+
+                List<LabeledIntent> labeledIntentList = new ArrayList<>();
```
Comment (vuk) [inline src/android/nl/xservices/plugins/SocialSharing.java]: Just noticed this line gives "diamond operator not supported error"

**Pull request merged**